File: JavaScript/Two_Sum.js
// Prompt:

// Answer:
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
  const hashtable = {};
  for (let i = 0; i < nums.length; i++) {
    let complement = target - nums[i];

    if (hashtable.hasOwnProperty(complement)) {
      return [hashtable[complement], i];
    }

    hashtable[nums[i]] = i;
  }
  return null;
};

File: Python/Two_Sum.py
# Prompt:

# Answer:
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = {}

        for idx, value in enumerate(nums):
            complement = target - value

            if complement in hashtable:
                return [hashtable[complement], idx]

            hashtable[value] = idx
